test/e2e/framework/types/runner.go

```diff
@@ -20,9 +21,20 @@ func NewRunner(t *testing.T, job *Job) *Runner {
 	}
 }
 
-func (r *Runner) Run() {
+func (r *Runner) Run(ctx context.Context) {
 	if r.t.Failed() {
 		return
 	}
-	require.NoError(r.t, r.Job.Run())
+	runComplete := make(chan struct{})
+
+	go func() {
+		require.NoError(r.t, r.Job.Run())
+		runComplete <- struct{}{}
+	}()
+	select {
+	case <-ctx.Done():
+		r.t.Fatal("Timeout")
```

Comment:
add that this is a go test runner timeout, could be ambiguous with sporadic infra timeouts that could happen

Reply:
+1. The message you leave in `t.Fatal` is the message you will read in a cryptic jumble of log output. Be kind to yourself and always make it as descriptive as you can.